crates/ty_python_semantic/src/types/function.rs

```diff
@@ -731,6 +731,11 @@ impl<'db> FunctionType<'db> {
         self.literal(db).implementation_deprecated(db)
     }
 
+    /// Returns if this function is a class method.
+    pub(crate) fn is_class_method(self, db: &dyn Db) -> bool {
+        self.has_known_decorator(db, FunctionDecorators::CLASSMETHOD) || self.name(db) == "__new__"
+    }
```

Comment:
I added a `is_classmethod` method on `FunctionType` recently (https://github.com/astral-sh/ruff/pull/20190). Note that `__new__` is not a classmethod (https://github.com/astral-sh/ruff/pull/20190#discussion_r2313195861). I suggest we remove this, and use `is_classmethod` below.